foursight_core/app_utils.py

```diff
@@ -334,6 +574,191 @@ def trim_output(cls, output, max_size=100000):
+        the updating of which does not update the modified time; so the code (get_lambda_last_modified)
+        to get the last modified time of the lambda needs to look first at this tag and take that
+        if it exists before looking at the real last modified time.
+        """
+        if not lambda_name or lambda_name.lower() == "current":
+            lambda_name = os.environ.get("AWS_LAMBDA_FUNCTION_NAME")
+            if not lambda_name:
+                return False
+        try:
+            boto_lambda = boto3.client("lambda")
+            lambda_info = boto_lambda.get_function(FunctionName=lambda_name)
+            if lambda_info:
+                lambda_arn = lambda_info["Configuration"]["FunctionArn"]
+                lambda_tags = boto_lambda.list_tags(Resource=lambda_arn)["Tags"]
+                lambda_last_modified_tag = lambda_tags.get("last_modified")
+                if not lambda_last_modified_tag:
+                    lambda_last_modified = lambda_info["Configuration"]["LastModified"]
+                    boto_lambda.tag_resource(Resource=lambda_arn, Tags={"last_modified": lambda_last_modified})
+                lambda_description = lambda_info["Configuration"]["Description"]
+                if not lambda_description:
+                    lambda_description = "Reload"
+                else:
+                    if lambda_description.endswith("."):
+                        lambda_description = lambda_description[:-1]
+                    else:
+                        lambda_description = lambda_description + "."
+                logger.warn(f"Reloading lambda: {lambda_name}")
+                boto_lambda.update_function_configuration(FunctionName=lambda_name, Description=lambda_description)
+                logger.warn(f"Reloaded lambda: {lambda_name}")
+        except Exception as e:
+            logger.warn(f"Error reloading lambda ({lambda_name}): {e}")
+        return False
+
+    def get_lambda_last_modified(self, lambda_name: str = None) -> str:
+        """
+        Returns the last modified time for the given lambda name.
+        See comments in reload_lambda on this.
+        """
+        lambda_current = False
+        if not lambda_name or lambda_name.lower() == "current":
+            lambda_name = os.environ.get("AWS_LAMBDA_FUNCTION_NAME")
+            if not lambda_name:
+                return None
+            lambda_current = True
+        if lambda_current:
+            if self.lambda_last_modified:
+                return self.lambda_last_modified
+        try:
+            boto_lambda = boto3.client("lambda")
+            lambda_info = boto_lambda.get_function(FunctionName=lambda_name)
+            if lambda_info:
+                lambda_arn = lambda_info["Configuration"]["FunctionArn"]
+                lambda_tags = boto_lambda.list_tags(Resource=lambda_arn)["Tags"]
+                lambda_last_modified_tag = lambda_tags.get("last_modified")
```

Comment:
Does the original foursight deployment do this? Or will this not work until you do the first reload and push the tag?

Reply:
Not sure what you mean. You mean does the deployment create a tag initially? Yes.